src/cmd/networkmapper/visualize/mapper-visualize.go

```diff
@@ -0,0 +1,240 @@
+		panic(err)
+	}
+	graph.SetRankDir(cgraph.LRRank)
+	return &Visualizer{
+		Graphviz:    g,
+		graph:       graph,
+		nodeCache:   make(map[string]*cgraph.Node, 0),
+		graphFormat: format,
+	}
+}
+
+func (v *Visualizer) addToCache(nodeName string) error {
+	if _, ok := v.nodeCache[nodeName]; !ok {
+		node, err := v.graph.CreateNode(nodeName)
+		if err != nil {
+			return err
+		}
+		node.SetMargin(0.2)
+		v.nodeCache[nodeName] = node
+	}
+	return nil
+}
+
+func (v *Visualizer) populateNodeCache(serviceIntents []mapperclient.ServiceIntentsUpToMapperV017ServiceIntents) error {
+	for _, service := range serviceIntents {
+		clientName := fmt.Sprintf("%s.%s", service.Client.Name, service.Client.Namespace)
+		if err := v.addToCache(clientName); err != nil {
+			return err
+		}
+		for _, intent := range service.Intents {
+			targetServiceName := v.formatTargetServiceName(service.Client.Namespace, intent)
+			if err := v.addToCache(targetServiceName); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (v *Visualizer) buildEdges(serviceIntents []mapperclient.ServiceIntentsUpToMapperV017ServiceIntents) error {
+	for _, service := range serviceIntents {
+		clientName := fmt.Sprintf("%s.%s", service.Client.Name, service.Client.Namespace)
+		for _, intent := range service.Intents {
+			targetNameWithNS := v.formatTargetServiceName(service.Client.Namespace, intent)
+			_, err := v.graph.CreateEdge(
+				fmt.Sprintf("%s to %s", clientName, targetNameWithNS),
+				v.nodeCache[clientName],
+				v.nodeCache[targetNameWithNS])
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+var VisualizeCmd = &cobra.Command{
+	Use:   "visualize",
+	Short: "Visualize an access graph for network mapper intents using go-graphviz",
+	Args:  cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return mapperclient.WithClient(func(c *mapperclient.Client) error {
+			namespacesFilter := viper.GetStringSlice(NamespacesKey)
+			format := viper.GetString(GraphFormatKey)
+			graphFormat, err := getGraphvizFormat(format)
+			if err != nil {
+				return err
+			}
+
+			outFile := viper.GetString(config.OutputPathKey)
+			servicesIntents, err := c.ServiceIntents(context.Background(), namespacesFilter)
+			if err != nil {
+				return err
+			}
+			visualizer := NewVisualizer(graphFormat)
+			defer func() {
+				if err := visualizer.graph.Close(); err != nil {
+					panic(err)
+				}
+				if err := visualizer.Close(); err != nil {
+					panic(err)
+				}
+			}()
+
+			if err := visualizer.populateNodeCache(servicesIntents); err != nil {
+				return err
+			}
+			if err := visualizer.buildEdges(servicesIntents); err != nil {
+				return err
+			}
+			if err := visualizer.RenderFilename(visualizer.graph, graphFormat, outFile); err != nil {
+				return err
+			}
+			if err := visualizer.addWatermark(outFile); err != nil {
+				return err
+			}
+
+			output.PrintStdout("Exported graph as %s format to path %s", format, outFile)
```

Comment:
Informational/error output to user should go to stderr - stdout is for actual output (e.g. if you would output the image itself)